File: 0001_0100/0019/0019.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    t = [(ord(x[0]) - ord('A'), int(x[1]) - 1) for x in f.readline().split()]
    a = 0
    for i in range(8):
        if i in (t[0][0], t[1][0]):
            a += 8
        else:
            for j in range(8):
                if j in (t[0][1], t[1][1]) or abs(t[0][0] - i) == abs(t[0][1] - j) \
                   or abs((t[2][0] - i) * (t[2][1] - j)) == 2 or (i, j) == t[2]:
                    a += 1
    q.write(str(a - 3))

File: 0001_0100/0019/0019.go
package main

import (
	"fmt"
	"os"
)

func abs(x int8) int8 {
	y := x >> 7
	return (x ^ y) - y
}

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	get := func() (int8, int8) {
		var s string
		fmt.Fscan(f, &s)
		return int8(s[0] - 'A'), int8(s[1] - '1')
	}
	qx, qy := get()
	rx, ry := get()
	kx, ky := get()
	n := 0
	for i := int8(0); i < 8; i++ {
		if i == qx || i == rx {
			n += 8
		} else {
			for j := int8(0); j < 8; j++ {
				if j == qy || j == ry || abs(qx-i) == abs(qy-j) || abs((kx-i)*(ky-j)) == 2 || (i == kx && j == ky) {
					n++
				}
			}
		}
	}
	fmt.Fprint(q, n-3)
}
